pkg/validation-webhook/snapshot.go

```diff
@@ -55,24 +55,24 @@ func admitSnapshot(ar v1.AdmissionReview) *v1.AdmissionResponse {
 	deserializer := codecs.UniversalDeserializer()
 	switch ar.Request.Resource {
 	case SnapshotV1Beta1GVR:
```

Comment:
do we need new case statements for v1?

Reply:
I think we only need to handle v1beta1 here as we use oneof in the v1 manifest.

Reply:
The default case throws an error, is that going to be problematic?

Reply:
Probably safer just to ignore the error and log a message?

Reply:
or add a case statement but don't validate anything.

Is there anything that we validate for v1beta that still needs to be validated for v1?

Reply:
For invalid v1 objects, we get the following message from the "oneOf" restriction:
```
kubectl create -f invalid-snapshot_v1.yaml
The VolumeSnapshot "invalid-snapshot-demo" is invalid: : Invalid value: "": "spec.source" must validate one and only one schema (oneOf). Found 2 valid alternatives
```

For invalid v1beta1 objects, we get the following message from the webhook:
```
kubectl create -f invalid-snapshot_beta.yaml
Error from server: error when creating "invalid-snapshot_beta.yaml": admission webhook "validation-webhook.snapshot.storage.k8s.io" denied the request: only one of Spec.Source.PersistentVolumeClaimName = pvc and Spec.Source.VolumeSnapshotContentName = vsc should be set
```

Reply:
We do need to return a *v1.AdmissionResponse here.  So we can't just do nothing.   We need to specify Allowed to true or false.
```
        reviewResponse := &v1.AdmissionResponse{
                Allowed: true,
                Result:  &metav1.Status{},
        }
```

Reply:
@yuxiangqian can you take a look?

Reply:
Hm so under the webhook config, we only have rules for v1beta: https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/webhook-example/admission-configuration-template#L9

I guess that means v1 objects will not go through the webhook, and actually at some point after v1beta deprecation, we don't need the webhook?

Reply:
I think so.  @yuxiangqian can you confirm?

Reply:
I am afraid we need to add support for v1 resources as well.
@msau42  to your point, currently the webhook server will **ONLY** receive v1beta1 object related admission requests, and its configured [here](https://github.com/kubernetes-csi/external-snapshotter/blob/master/deploy/kubernetes/webhook-example/admission-configuration-template#L9)

this would need to be extended to also cover v1 in this or a separated PR.
if only handle v1beta1 objects, we will not have problem with **OneOf** semantic as v1 schema guarantees that, however it might cause problems with other validation rules, i.e., immutable checks [here](https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/validation-webhook/snapshot.go#L158) will not apply to v1 resources.

Reply:
I think it makes sense to introduce v1 admission along with v1beta1 -> v1(GA) in the validation webhook. The change should not be very big. For v1 objects, we can ignore **OneOf** checks and leave it to schema level validation, but to keep it consistent, immutable checks/other validation checks(for [example](https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/utils/util.go#L140)) should be implemented on v1 objects.

Reply:
> The default case throws an error, is that going to be problematic?
@yuxiangqian Can you answer the above question from Michelle?